src/types.ts

```diff
@@ -0,0 +1,73 @@
+import { IFlagArgument, IFlagOptions, semver } from "../deps.ts";
+
+export const releases = [
+  "patch",
+  "minor",
+  "major",
+  "pre",
+  "prepatch",
+  "preminor",
+  "premajor",
+  "prerelease",
+];
+
+export function validateRelease(value: string): boolean {
+  return releases.includes(value);
+}
+
+export function releaseType(
+  option: IFlagOptions | string,
+  arg: IFlagArgument,
+  value: string,
+): semver.ReleaseType {
+  if (!validateRelease(value)) {
+    throw new Error(
+      `Option ${
+        typeof option === "string" ? option : `--${option.name}`
+      } must be a valid release type but got: ${value}.\nAccepted values are ${
+        releases.join(", ")
+      }.`,
+    );
+  }
+  return value as semver.ReleaseType;
+}
+
+export function validateVersion(value: string): boolean {
+  return !!semver.valid(value);
+}
+
+export function versionType(
+  option: IFlagOptions | string,
+  arg: IFlagArgument,
+  value: string,
+): string {
+  if (!validateVersion(value)) {
+    throw new Error(
+      `Option ${
+        typeof option === "string" ? option : `--${option.name}`
+      } must be a valid version but got: ${value}.\nVersion must follow Semantic Versioning 2.0.0.`,
+    );
+  }
+  return value;
+}
+
+export function validateURL(value: string): boolean {
+  const urlRegex =
```

Comment:
This is perfectly fine. Just brainstorming...

What if we just made a request to the given url? If it's a virus or anything malicious, this would be run on the user's computer anyway...

Reply:
I didn't quite understand what you meant. 🤔
It's just a matter of checking that the field is indeed an url, making a request is not required.